pkg/server/statement_diagnostics_requests.go

```diff
@@ -259,3 +265,110 @@ func (s *statusServer) StatementDiagnostics(
 
 	return response, nil
 }
+
+// getStatementBundle retrieves the statement bundle with the given id and
+// writes it out as an attachment. Note this function assumes the user has
+// permission to access the statement bundle.
+func (s *adminServer) getStatementBundle(ctx context.Context, id int64, w http.ResponseWriter) {
+	row, err := s.internalExecutor.QueryRowEx(
+		ctx, "admin-stmt-bundle", nil, /* txn */
+		sessiondata.NodeUserSessionDataOverride,
+		"SELECT bundle_chunks FROM system.statement_diagnostics WHERE id=$1 AND bundle_chunks IS NOT NULL",
+		id,
+	)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if row == nil {
+		http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+		return
+	}
+	// Put together the entire bundle. Ideally we would stream it in chunks,
+	// but it's hard to return errors once we start.
+	chunkIDs := row[0].(*tree.DArray)
+	bundle, err := s.buildBundle(ctx, chunkIDs)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set(
+		"Content-Disposition",
+		fmt.Sprintf("attachment; filename=stmt-bundle-%d.zip", id),
+	)
+
+	_, _ = io.Copy(w, &bundle)
+}
+
+func (s *adminServer) buildBundle(
+	ctx context.Context, chunkIds *tree.DArray,
+) (bytes.Buffer, error) {
+	var bundle bytes.Buffer
+	for _, chunkID := range chunkIds.Array {
+		chunkRow, err := s.internalExecutor.QueryRowEx(
+			ctx, "admin-stmt-bundle", nil, /* txn */
+			sessiondata.NodeUserSessionDataOverride,
+			"SELECT data FROM system.statement_bundle_chunks WHERE id=$1",
+			chunkID,
+		)
+		if err != nil {
+			return bytes.Buffer{}, errors.Wrap(err, "internal server error")
+		}
+		if chunkRow == nil {
+			return bytes.Buffer{}, errors.Newf("Not Found")
```

Comment:
Maybe change error message to "No bundle chunks found for id: %d"

Reply:
done